File: project/3a/frontend/mobile/core/orders/interface/order.ts
export type OrderStatus =
    | 'PENDING'
    | 'IN_PREPARATION'
    | 'READY'
    | 'DELIVERED'
    | 'CANCELLED';

// One ordered unit. Two burgers with different instructions come as two
// entries with different `notes`. The waiter UI groups visually by
// (itemName, notes) when rendering existing orders.
export interface OrderItem {
    id: number;
    itemName: string;
    notes: string | null;
}

export interface Order {
    id: number;
    restaurantId: number;
    customerId?: number | null;
    customerName?: string | null;
    tableNumber: number;
    status: OrderStatus;
    notes: string | null;
    totalAmount: number;
    priority: number;
    requestedArrivalTime?: string | null;
    arrivalMessage?: string | null;
    cancellationReason?: string | null;
    cancelledAt?: string | null;
    createdAt: string;
    updatedAt: string;
    items: OrderItem[];
}

export interface CreateOrderItemInput {
    itemName: string;
    notes?: string | null;
}

export interface CreateOrderInput {
    restaurantId: number;
    tableNumber: number;
    customerId?: number | null;
    customerName?: string | null;
    totalAmount?: number | null;
    notes?: string | null;
    items: CreateOrderItemInput[];
}

export type EtaMode = 'WALKING' | 'DRIVING';

export interface OrderEta {
    orderId: number;
    restaurantId: number;
    mode: EtaMode;
    etaMinutes: number | null;
}

export type KitchenEventType = 'ORDER_CREATED' | 'ORDER_STATUS_CHANGED';

export interface KitchenEvent {
    type: KitchenEventType;
    order: Order;
    timestamp: string;
}

File: project/3a/backend/OrderService/src/main/java/com/clickmunch/OrderService/entity/Order.java
package com.clickmunch.OrderService.entity;

import java.math.BigDecimal;
import java.time.LocalDateTime;

import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

import lombok.Data;

@Data
@Table("orders")
public class Order {
    @Id
    private Long id;
    private Long restaurantId;
    private Long customerId;
    private String customerName;
    private Integer tableNumber;
    private Long tableId;
    private OrderStatus status;
    private String notes;
    private BigDecimal totalAmount;
    private Integer priority;
    private LocalDateTime requestedArrivalTime;
    private String arrivalMessage;
    private String cancellationReason;
    private LocalDateTime cancelledAt;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}
